core/services/ocr2/plugins/ocr2keeper/evm21/transmit/cache.go

```diff
@@ -0,0 +1,80 @@
+package transmit
+
+import (
+	"encoding/hex"
+	"sync"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg/v3/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+)
+
+// transmitEventCache holds a ring buffer of the last visited blocks (transmit block),
```

Comment:
I'm not totally sure but it seems like ring buffers are used quite a bit in core/keepers; should we create a single ring buffer type, rather than having different types that behave as ring buffers?